caikit/runtime/servicers/global_train_servicer.py

```diff
@@ -177,57 +165,49 @@ def Train(self, request, context, *_, **__) -> TrainingJob:
 
     def run_training_job(
         self,
-        request,
-        model,
-        training_id,
-        training_output_dir,
-        context: ServicerContext,
-        wait=False,
+        request: Union[ProtoMessageType, DataBase],
+        module: Type[ModuleBase],
+        training_output_dir: str,
+        *,
+        context: Optional[ServicerContext] = None,
+        wait: bool = False,
+        **kwargs,
     ) -> TrainingJob:
         """Builds the request dict for calling the train function asynchronously,
-        then returns the thread id"""
+        then returns the thread id
 
-        # Figure out where this model will be saved
-        model_name = request.model_name
-        model_path = self._get_model_path(training_output_dir, model_name, training_id)
-
-        # Build the full set of kwargs for the train and save call
-        kwargs = {
-            "module_class": model,
-            "model_path": model_path,
-            **build_caikit_library_request_dict(request, model.TRAIN_SIGNATURE),
-        }
-
-        # If running with a subprocess, set the target, events and args accordingly
-        target = (
-            SubProcessTrainSaveExecutor()
-            if self.use_subprocess
-            else LocalTrainSaveExecutor()
-        )
-        log.debug2(
-            "Training with %s",
-            "SUBPROCESS" if self.use_subprocess else "MAIN PROCESS",
-        )
+        Args:
+            request (Union[ProtoMessageType, DataBase]): The message that
+                stimulated this request
+            module (Type[ModuleBase]): The module class to train
+            training_output_dir (str): The base directory where trained models
+                should be saved
 
-        # start training asynchronously
-        thread_future = self.run_async(
-            runnable_executor=target,
-            kwargs=kwargs,
-            model_name=model_name,
-            model_path=model_path,
-        )
+        Kwargs:
+            context (Optional[ServicerContext]): The grpc context for the
+                request if called from a grpc handler
+            wait (bool): Whether or not to block until the training is complete
+
+        Returns:
+            training_job (TrainingJob): The job handle for the training with the
+                job's ID and the model's name
+        """
 
-        self.training_map[training_id] = thread_future
+        # Figure out where this model will be saved
+        model_path = self._get_model_path(training_output_dir, request.model_name)
+
+        # Build the full set of kwargs for the train call
+        kwargs.update(
+            {
+                "module": module,
+                "save_path": model_path,
+                "save_with_id": True,
+                **build_caikit_library_request_dict(request, module.TRAIN_SIGNATURE),
```

Comment:
doesn't this assume that the `request` type is a `ProtoMessageType`? How will this work with REST calls?

Reply:
On the inference side, the call to `Predict` does the `build_caikit_library_request_dict`, which creates kwargs and passes to `predict_model`.

`predict_model` takes in these kwargs which are used to actually call the `model_run_fn`. Should we do the same thing here?

Reply:
Hm, good point! Let me see if that can be refactored to match GPS better

Reply:
I looked into this a bit and I think I'm going to punt it. Shifting this out of `run_training_job` requires `ModelTrainServicer` to also use `build_caikit_library_request_dict` which seems to have some nuance. Due to the `DataStreamSource` stuff, this is a bit complex and does need to be coerced into the data model object to work cleanly rather than just a pile of `**kwargs`. I think we'll want to dig into this as part of getting train working with REST.

Reply:
Sounds good!